cmd/reposervercontroller/main.go

```diff
@@ -93,7 +96,9 @@ func main() {
 		setupLog.Error(err, "unable to set up ready check")
 		os.Exit(1)
 	}
-
+	if err := resource.CreateRepoServerCustomResource(context.Background(), config); err != nil {
+		setupLog.Error(err, "problem creating repository server custom resource")
+	}
```

Comment:
I am not using the same env, that env variable is used to decide whether CRDs should be installed using helm or should be created using go code. We are currently not putting the repo server CRD under helm chart since we are hiding it behind a flag. The `CreateRepoServerCustomResource` mandatorily creates the CRD for repository server whenever the repo server controller is started. We can move to one flag later when we want to distribute repo server CRD using helm chart. WDYT?

Reply:
The Repo server CRD will only be created when the repo server controller is deployed since we have added the logic of creating repo server CRD under `cmd/reposervercontroller/main.go`

Reply:
> The CreateRepoServerCustomResource mandatorily creates the CRD for repository server whenever the repo server controller is started(i.e whenever we deploy the controller). We can move to same flag later when we want to distribute repo server CRD using helm chart. WDYT?

Is there a reason why we decided to not distribute the CRD using helm as of now? Some users might not be able to use this feature if they can not give the repo server controller permissions (authz) to manage the CRDs. And that was the reason, for main CRDs we decided that we should give users the options. 

Even if we keep the repo server controller behind feature flag, I think we can have CRD in helm template as well right? Its deployment would  depend on that feature flag.

Reply:
alright, I have made the changes. PTAL